mobius-android/src/main/java/com/spotify/mobius/android/EffectLiveData.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.lifecycle.Observer;
+
+/**
+ * An interface for an object that acts like an Android LiveData, except it will only emit each
```

Comment:
This interface as defined is really only about observing. So it's more like a `LifecycleAwareObservable` or something along those lines. The main method missing (apart from one observable-related mentioned below) from `LiveData` is `getValue()`. I don't think, given the intended use of this type, that we should add that. 

The semantics around queueing/replaying T:s are defined in an implementation as things stand today. I'm not sure if that should somehow be reflected in the naming and documentation. I guess the only reason for introducing this interface is to have a place to define the queuing/replaying semantics? I think we should give the name of this some more thought. Some ideas:

- QueuingLiveData
- LifecycleAwareObservable
- ViewEffectQueue (if it's made very un-generic and only supports the single use-case we use it for today).

Reply:
The reason I had to write this interface is to expose an observable-only live-data like object (basically hide the post/postTransient calls) - and unfortunately, the way Google has defined Live Data it's an abstract class, not an interface, so I couldn't use that. 

Yea, this shouldn't have getValue since that would then mean exposing the Accumulator, which the rest of the work here tried not to do. I'll rename it to something better 👍